java/src/main/java/org/eclipse/ditto/client/messaging/internal/WebSocketMessagingProvider.java

```diff
@@ -364,15 +356,32 @@ public void onConnected(final WebSocket websocket, final Map<String, List<String
             if (!subscriptionMessages.isEmpty()) {
                 LOGGER.info("Client <{}>: Subscribing again for messages from backend after reconnection",
                         sessionId);
-                subscriptionMessages.values().forEach(this::emit);
+                CompletableFuture<Boolean> isReconnecting = new CompletableFuture<>();
+                Runnable checkTask = () -> {
+                    if (!reconnecting.get()) {
+                        isReconnecting.complete(true);  // Complete the future if flag is true
+                    }
+                };
+                ScheduledFuture<?> future = connectExecutor.scheduleAtFixedRate(checkTask, 0, 20, TimeUnit.MILLISECONDS);
+                try {
+                    if (Boolean.TRUE.equals(isReconnecting.get(160, TimeUnit.MILLISECONDS))) { // Ensures 4 retries of the scheduleAtFixedRate method.
```

Comment:
@kalinkostashki I don't get which 160ms would ensure only 4 retries.
As I understand `scheduleAtFixedRate`, the `period` parameter is static - so after each execution, the next one will be 20ms after.  
So after 4 executions you are only slightly above 80ms.

Either way - I propose to configure the `20` magic number as constant with a good name - and have another constant like `MIN_RECONNECTING_CHECK_TRIES` - and calculate the max time to wait based on those 2 constants.  
Just to get magic numbers out of the code ..

Reply:
@thjaeckle 
What I read from the java doc about scheduleAtFixedRate() in ScheduledExecutorService:
Submits a periodic action that becomes enabled first after the given initial delay, and subsequently with the given period; that is, executions will commence after initialDelay, then initialDelay + period, then initialDelay + 2 * period, and so on.

From this I thought that it will trigger at 0, 20, 40, 80, 160 -> hence my comment.

Reply:
No, I read it as:
* the first execution is at `initialDelay` (0ms in our case)
* the second at `initialDelay + period` (20ms in our case)
* the third at `initialDelay + 2*period` (40ms in our case)
* the 4th at `initialDelay + 3*period` (60ms in our case)

It doesn't mention that `period` is doubled on each execution - basically every 20ms the task is executed.

Reply:
oops, yes, that makes more sense :smiley: 
I will take care of the other comments.
Thanks for the hints :)